app/policies/assignment_policy.rb

```diff
@@ -42,6 +42,6 @@ def can_be_graded?
   end
 
   def show_grades?
-    assignment.graded? && Time.current > assignment.deadline
+    assignment.graded? && Time.current > assignment.deadline && assignment.grades_published?
```

Comment:
This will make all the existing assignments in the system unavailable

Reply:
Sorry, but I was unable to figure out how would it make all the existing assignments in the system unavailable? 
The file show.html.erb for groups contains the following code -
```ruby
                <% if policy(assignment).show_grades? && project_assignment.present? %>
                  <div class="groups-assignment-details">
                    <div class="groups-no-assignment-submission-text">
                      <h6>
                        <%= ProjectDecorator.new(project_assignment).grade_str %>
                      </h6>
                      <div class="groups-expanded-assignment-name">
                        <p>(<%= ProjectDecorator.new(project_assignment).remarks_str %>)</p>
                      </div>
                    </div>
                  </div>
                <% else %>
                  <div class="groups-assignment-details">
                    <h6 class="groups-no-assignment-submission-text"><%= t("not_available") %></h6>
                  </div>
                <% end %>
```
So, if the assignment's policy show_grades is false, then the grade and remarks would be shown as "not_available".
Maybe, I am missing out on something here. Can you please point out where the mistake is?

Reply:
I think `assignment.grades_published?` will be default false for all the previous assignments in our system that's why.

Reply:
yes but even then why would the existing assignments become unavailable? The grades and remarks section should show "not available".

Reply:
Yeah but that's not the expected behaviour, because already graded assignments should continue to show the marks as before.
According to me, the new feature shouldn't suppress our previous behaviour.